upgrades/0.0.105.add-node-volume-encryption/upgrade.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	survey "github.com/AlecAivazis/survey/v2"
+	"github.com/google/uuid"
+	"github.com/oslokommune/okctl-upgrade/upgrades/0.0.105.add-node-volume-encryption/pkg/eksctl"
+	"github.com/oslokommune/okctl-upgrade/upgrades/0.0.105.add-node-volume-encryption/pkg/lib/cmdflags"
+	"github.com/oslokommune/okctl-upgrade/upgrades/0.0.105.add-node-volume-encryption/pkg/lib/logging"
+	"github.com/oslokommune/okctl-upgrade/upgrades/0.0.105.add-node-volume-encryption/pkg/lib/manifest/apis/okctl.io/v1alpha1"
+)
+
+func upgrade(context Context, flags cmdflags.Flags, cluster v1alpha1.Cluster) error {
+	log := context.logger
+
+	if !flags.Confirm {
+		continueUpgrade, err := prompt(continuationMessage)
+		if err != nil {
+			return fmt.Errorf("prompting for continuation: %w", err)
+		}
+
+		if !continueUpgrade {
+			return errors.New("aborted by user")
+		}
+	}
+
+	log.Debug("Retrieving nodegroup names")
+
+	originalNodegroupNames, err := eksctl.GetNodeGroupNames(cluster.Metadata.Name)
+	if err != nil {
+		return fmt.Errorf("acquiring nodegroup names: %w", err)
+	}
+
+	log.Debugf("Found nodegroups %v\n", originalNodegroupNames)
+
+	clusterVersion, err := eksctl.GetClusterVersion(cluster.Metadata.Name)
+	if err != nil {
+		return fmt.Errorf("acquiring cluster version: %w", err)
+	}
+
+	log.Debugf("Found cluster version %s\n", clusterVersion)
+
+	log.Debug("Generating eksctl cluster configuration")
+
+	cfg, err := eksctl.GenerateClusterConfig(
+		cluster,
+		generateNodegroupNames(cluster.Metadata.Region, clusterVersion, generateRandomString),
+	)
+	if err != nil {
+		return fmt.Errorf("generating cluster config: %w", err)
+	}
+
+	cfg, err = configLogPrinter(log, cfg)
+	if err != nil {
+		return fmt.Errorf("debug printing cluster config: %w", err)
+	}
+
+	log.Debug("Updating nodegroups")
+
+	err = eksctl.UpdateNodeGroups(cfg, flags.DryRun)
```

Comment:
We're creating a node group here (using eksctl create) rather than updating an existing one, should the function be called CreateNodeGroup?